### ISSUE TITLE
CPU load spikes to 100 percent

### ISSUE THREAD

**Issue by mboehm21** [neutrinolabs/xrdp#1907]
I use Guacamole to open an XRDP-Session into an Ubuntu 18.04 container.

When I open Firefox and it tries to load an unresponsive website, the CPU usage of xrdp and Xorg spikes and the connection becomes unresponsive until Firefox timed out:

```
 PID  USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND
 4374 root      20   0   48924  26124   9236 R  99.7  0.1   0:16.51 xrdp                                                                                                                                       
 4377 user      20   0 1566700 110512  43284 S  28.6  0.3   0:06.60 Xorg                                                                                                                                       
```

```
root@desktop:/# cat /etc/os-release 
NAME="Ubuntu"
VERSION="18.04.5 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.5 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
root@desktop:/# xrdp --version

xrdp: A Remote Desktop Protocol server.
Copyright (C) Jay Sorg 2004-2014
See http://www.xrdp.org for more information.
Version 0.9.5
```

Is this a known issue and how can I troubleshoot it? Thanks.

**Comment by metalefty** [comment]
You're using an ancient version of xrdp. We released 7 new versions after 0.9.5. You should test it with the latest version and the issue still occurs with the latest version, report it again.

**Comment by mboehm21** [comment]
Thanks, it seems that the upstream Docker image I'm using has this old version for 18.04. I'll try the 20.04 version, that has xrdp 0.9.12.
